### PR TITLE
Implement nanpercentile function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
## Description
Adds keras.ops.nanpercentile, which computes the q-th percentile of the input tensor along a specified axis while ignoring NaN values.
Supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

> **Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -5956,6 +5956,91 @@ def nanmin(x, axis=None, keepdims=False):
...
+                output_shape = (q.shape[0],) + output_shape
+        elif isinstance(q, (list, tuple)) and len(q) > 1:
+            output_shape = (len(q),) + output_shape
+
+        if backend.standardize_dtype(x.dtype) == "int64":
+            dtype = backend.floatx()
+        else:
+            dtype = dtypes.result_type(x.dtype, float)
+
+        return KerasTensor(output_shape, dtype=dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:5989]: 1.Since q is constrained to be a scalar or 1D tensor (ndim <= 1), the current shape inference logic using q.shape[0] is sufficient and consistent with the intended API. Therefore, no change is made in this part.
2. Updated
3. Updated.

File: keras/src/backend/numpy/numpy.py [hunk 2]
```diff
@@ -1093,6 +1093,17 @@ def nanmin(x, axis=None, keepdims=False):
     return np.nanmin(x, axis=axis, keepdims=keepdims)
 
 
+def nanpercentile(x, q, axis=None, method="linear", keepdims=False):
+    x = convert_to_tensor(x)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1097]: Updated

File: keras/src/backend/torch/numpy.py [hunk 3]
```diff
@@ -1463,6 +1463,12 @@ def nanmin(x, axis=None, keepdims=False):
     )
 
 
+def nanpercentile(x, q, axis=None, method="linear", keepdims=False):
+    x = convert_to_tensor(x)
+    q = convert_to_tensor(q) / 100.0
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py]: Updated

**Review by hertschuh — APPROVED** [review]
Thanks!

But I'm realizing now that we don't even have `ops.numpy.percentile` right now.

**Comment by shashaka** [comment]
I’ll take a look and add a implementation for ops.numpy.percentile.
